### PR TITLE
Fix kernel header downloading on openSUSE & clean up broken kernel headers

### PR DESCRIPTION
**Pull request by ISauve**
### What does this PR do?

Updates the `nikos` dependency to pull in https://github.com/DataDog/nikos/pull/28, which includes a couple fixes for kernel header downloading on openSUSE.

In addition, adds a check during kernel header validation that the `linux/types.h` file can be found, and if it can't, delete those headers. This is to remove kernel headers with broken symlinks.

### Describe how to test/QA your changes

1) Spin up a vagrant openSUSE box, i.e. [generic/opensuse42](https://app.vagrantup.com/generic/boxes/opensuse42). Make sure that the repos in `/etc/zypp/repos.d` are type `yast2` (if you `cat` the file, `type` should be one of the entries in it). If they're not `yast2`, try a newer openSUSE box
2) Enable runtime compilation and force kernel header downloading to occur:
```
system_probe_config:
  enabled: true
  enable_runtime_compiler: true
  kernel_header_dirs: /dev/null
```
3) Start the system-probe and check the logs in order to verify that a. kernel header downloading occurred, b. both the `kernel-default-devel-[kernel_version]` and `kernel-devel-[kernel_version]` packages were downloaded, and c. runtime compilation succeeded:
```
| Downloading kernel headers for target distribution opensuse, release 42.3, kernel 4.4.179-99-default
...
| Looking for package kernel-default-devel-4.4.179-99
| Found package kernel-default-devel-4.4.179-99.1.x86_64
| Downloading package
| Downloading: https://ftp5.gwdg.de/pub/opensuse/discontinued/update/leap/42.3/...
| successfully downloaded kernel-default-devel-4.4.179-99
| Looking for package kernel-devel-4.4.179-99
| Found package kernel-devel-4.4.179-99.1.noarch
| Downloading package
| Downloading: https://ftp5.gwdg.de/pub/opensuse/discontinued/update/leap/42.3/...
| successfully downloaded kernel headers to /var/tmp/datadog-agent/system-probe-4/kernel-headers
| initialized ebpf conntrack
```

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/util/kernel/find_headers.go [hunk 1]
```diff
@@ -85,6 +87,23 @@ func GetKernelHeaders(headerDirs []string, headerDownloadDir, aptConfigDir, yumR
 	}
 	log.Debugf("unable to find downloaded kernel headers: %s", err)
 
+	if strings.Contains(err.Error(), "linux/types.h missing") {
```
Comment (brycekahle) [inline pkg/util/kernel/find_headers.go:90]: Can we use a sentinel error for this, using `errors.New()` and checking for it using `errors.Is()`? Checking for strings is prone to be brittle.

File: pkg/util/kernel/find_headers.go [hunk 2]
```diff
@@ -110,12 +135,32 @@ func validateHeaderDirs(hv Version, dirs []string) error {
...
+	}
+
+	log.Debugf("valid kernel headers found in %v", dirs)
+	return nil
+}
+
+func containsLinuxTypesHFile(dir string) bool {
+	path := filepath.Join(dir, "include/linux/types.h")
+	f, err := os.Open(path)
+	if os.IsNotExist(err) {
```
Comment (brycekahle) [inline pkg/util/kernel/find_headers.go:158]: This is how the docs recommend to check for this now. You will need to import `io/fs` too.
```suggestion
	if errors.Is(err, fs.ErrNotExist) {
```

File: pkg/util/kernel/find_headers.go [hunk 3]
```diff
@@ -110,12 +135,32 @@ func validateHeaderDirs(hv Version, dirs []string) error {
...
+		return false
+	}
+
+	defer f.Close()
```
Comment (brycekahle) [inline pkg/util/kernel/find_headers.go:162]: If an error other than `ErrNotExist` is returned, `f` will still be `nil`.
```suggestion
	if f != nil {
	    defer f.Close()
	}
```

**Comment by paulcacheux** [comment]
Hello ! Is there a plan to merge this for 7.34 ? Maybe only the nikos bump part ?

**Comment by ISauve** [comment]
@paulcacheux Yes, this is targeting 7.34

**Review by brycekahle — APPROVED** [review]

**Pull request merged**